### DISCUSSION TITLE
Clarifying Firmware Cut off

### DISCUSSION THREAD

**Discussion by zeroecho** [rudetrooper/Octoprint-Chituboard#23] [Q&A] [answered]
My Elegoo Mars 2 info panel shows I'm currently running EL3D-3.0.2, V4.4.3-c2_LCDC / F9.23

On the Mars 2 firmware page it shows the following options:

![Screen Shot 2022-04-07 at 10 45 18](https://user-images.githubusercontent.com/2373897/162171903-bbc0e686-8600-4c49-b86b-da17dc019f42.png)

I currently slice using Chitubox 1.9.0

Is V4.4.3 F9.23 before or after the encryption cut off? I only recently got the printer and did one firmware upgrade.

If I missed the boat (i.e. didn't realise the implications)

1) Is downgrading an option?
2) How would I slice after downgrading? I saw this exists https://github.com/CryptoCTB/LiberatingMarsCLI but I use Mac OS and guessing that is a windows only exe?

I just got a Pi Zero 2 W (ebay seller who didn't jack the price up, good people do exist!) and installed clean OctoPrint and installed from the plugin manager.

I tried to do the ssh instructions below after reboot and calling the .sh a second time, many of the links do not work and after the script finishes, I can no longer access the Pi Zero.

**Either from this Github or using wget https://raw.githubusercontent.com/rudetrooper/Octoprint-Chituboard/main/Chituboard.sh

Prepare for execution
sudo chmod +x ./Chituboard.sh

Execute
sudo bash ./Chituboard.sh**

3. Is the .sh script method still needed or is the plugin manager install sufficient?

Wanted to check the above 3 points before I crack open the case and start soldering

TIA!

**Comment by zeroecho** [comment, marked as answer]
So going to try it but more google-fu I think answers my first two points but not the last

1. Looks like yes from this thread https://www.reddit.com/r/ElegooMars/comments/otd7gi/chitubox_19latest_firmware_masterpost/
2. If I clicked releases, I'd have seen there is various compiled versions already https://github.com/CryptoCTB/LiberatingMarsCLI/releases/download/v0.3/LiberatingMarsCLI_macOS_x64
3. I still could do with clarity on this point

**Reply by zeroecho** [reply]
So to answer my own question in the hopes it helps others. 

3) was failing to run because my user was not 'pi' I flashed the image using the base OctoPi image but I used the raspberrypi image maker tool option to change from Pi to a different user for security purposes and the script is hard coded to use Pi.

Once that was fixed, I also fixed some of the broken link parts by running sudo apt-get update

Still not quite there (getting file cannot be read errors) but I think the hardware side is good.

Hoping the firmware is not my issue now but we'll see.

4) Anyone tried using lychee with this?

**Reply by ChristophCaina** [reply]
Hi, thanks for your loop replies... unfortunately, I run into the same issue (by mistake)...
I had an old Mars2 Pro... which was working fine... but I haven't used it quite a while...
Then - Updated the Chitu Slicer... got the information, that the printer can't read the file... did the firmware upgrade - and deleted the backup of the old one which was still on the usb drive... :( )

THEN - I decided to implement the octoprint to get the print status into my smarthome (homeassistant) ... and came accross this topic.
DAMN.... :-(
